pkg/openstack/instances_test.go

```diff
@@ -189,3 +190,65 @@ func TestSortNodeAddressesWithMultipleCIDRs(t *testing.T) {
 
 	executeSortNodeAddressesTest(t, addressSortOrder, want)
 }
+
+func Test_instanceIDFromProviderID(t *testing.T) {
+	type args struct {
+		providerID string
+	}
+	tests := []struct {
+		name           string
+		args           args
+		wantInstanceID string
+		wantRegion     string
+		wantErr        bool
+	}{
+		{
+			name: "it parses region & instanceID correctly from providerID",
+			args: args{
+				providerID: "openstack://us-east-1/testInstanceID",
+			},
+			wantInstanceID: "testInstanceID",
+			wantRegion:     "us-east-1",
+			wantErr:        false,
+		},
+		{
+			name: "it returns error in case of invalid providerID format with no region",
+			args: args{
+				providerID: "openstack://us-east-1-testInstanceID",
+			},
+			wantInstanceID: "",
+			wantRegion:     "",
+			wantErr:        true,
+		},
+		{
+			name: "it parses correct instanceID in case of extra slash protocal",
```

Comment:
"extra slash" makes it appear as a mistake. However, I _think_ that in our case the default region (which may be what we call "availability zone" in OpenStack) is the empty string (`""`). So the "region" is actually there; it's just that the region name has length zero.

If my speculation is correct (@dulek please keep me honest), I would change the test name to reflect that.

```suggestion
{
			name: "it parses correct instanceID in case the region name is the empty string",
}```

Reply:
My understanding is that region is not an AZ, region has whole new set of API services and only Keystone is shared. I don't think that matters too much here, your rename request seems valid, especially with the typo that's currently there. ;)

Reply:
Made requested changes